### PR TITLE
More G-module

### PR DESCRIPTION
**Pull request by fieker**

### CODE REVIEW

File: experimental/GModule/Cohomology.jl [hunk 1]
```diff
@@ -390,6 +403,10 @@ Base.:-(a::Generic.ModuleHomomorphism) = hom(domain(a), codomain(a), -mat(a))
 Base.zero(G::GrpAbFinGen) = G[0]
 Base.:-(M::GrpAbFinGenMap) = hom(domain(M), codomain(M), [-M(g) for g = gens(domain(M))], check = false)
 
+function Oscar.mat(M::FreeModuleHom{FreeMod{QabElem}, FreeMod{QabElem}})
```
Comment (fingolfin) [inline experimental/GModule/Cohomology.jl:406]: I think elsewhere this function exists and is called `matrix` (which also fits better in our general naming scheme). So perhaps instead change all uses of `mat` to `matrix`?

File: experimental/GModule/GModule.jl [hunk 2]
```diff
@@ -94,6 +94,26 @@ function irreducible_modules(::Type{AnticNumberField}, G::Oscar.GAPGroup)
...
+
+function irreducible_modules(::FlintIntegerRing, G::Oscar.GAPGroup)
+  z = irreducible_modules(G)
+  Z = GModule[]
+  for m in z
+    a = gmodule(CyclotomicField, m)
+    push!(Z, gmodule(ZZ, gmodule(QQ, a)))
+  end
+  return Z
+end
```
Comment (fingolfin) [inline experimental/GModule/GModule.jl]: Or maybe like this?
```suggestion
function irreducible_modules(::typeof(CyclotomicField), G::Oscar.GAPGroup)
  z = irreducible_modules(G)
  return [gmodule(CyclotomicField, m) for m in z]
end

function irreducible_modules(::FlintRationalField, G::Oscar.GAPGroup)
  z = irreducible_modules(CyclotomicField, G)
  return [gmodule(QQ, m) for m in z]
end

function irreducible_modules(::FlintIntegerRing, G::Oscar.GAPGroup)
  z = irreducible_modules(QQ, G)
  return [gmodule(ZZ, m) for m in z]
end
```

**Review by fingolfin — APPROVED** [review]

**Comment by fingolfin** [comment]
There's a conflict

**Comment by fingolfin** [comment]
You'll have to rebase on latest master, I think

File: experimental/GModule/Cohomology.jl (hunk 1 above)
Comment (fieker) [reply experimental/GModule/Cohomology.jl:406]: mat is still used (as of, I think yesterday) in AA? Otherwise, I agree totally.

**Pull request merged**